File: JDApp-Java/other/src/main/java/com/jd/other/network/BookService.java
package com.jd.other.network;


import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface BookService {

    @GET("/service/getIpInfo.php")
    Observable<String> getShop(@Query("ip") String ip);

}

File: JDApp-Kotlin/other/src/main/java/com/jd/other/network/BookService.kt
package com.jd.other.network

import com.jd.core.network.model.Response
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

interface BookService {
    @GET("/service/getIpInfo.php")
    fun getShop(@Query("ip") ip: String?): Observable<Response<String?>>?

    @GET("/service/getIpInfo.php")
    suspend fun getShop2(@Query("ip") ip: String?): Response<String?>
}